### ISSUE TITLE
Displaying Content From Modular Pages

### ISSUE THREAD

**Issue by LukeAskew** [getgrav/grav-plugin-simplesearch#93]
More of a question - the content from my site is mostly in modular pages. Searching matches the appropriate parent page, but the content is empty because the parent's content is non-existent.

I was able to work around this by commenting out [this block](https://github.com/getgrav/grav-plugin-simplesearch/blob/ef0cc71e690c267e264ea7e9f4e5d553ca2ad6f9/simplesearch.php#L181) so that modular pages are included in the results.

I modified my search result loop to display the parent page name but the modular page content.

Is there a better way to include the modular page's content in search results?

**Comment by flaviocopes** [comment]
It should search inside the modular content, but the page shown in the search should be the parent page, as a modular page "part" is not really a page, just a portion of it.

**Comment by LukeAskew** [comment]
Right, but the modular page actually contains the content, so if I want to display a meaningful search result, I need that content in the page model.

**Comment by mikegcox** [comment]
I cannot get SimpleSearch to give results on Modular pages at all.  I tried the mod by Luke but that did nothing either.  I want to be able to search the entire site.  Is there a setting I am missing?

```
enabled: true
built_in_css: false
display_button: true
min_query_length: 3
route: /search
template: simplesearch_results
filters:
filter_combinator: or
ignore_accented_characters: true
order:
  by: date
  dir: desc
```

**Comment by mikegcox** [comment]
My workaround in case someone is looking for an idea.  I created a series of pages with duplicate content using just the `default` template.  I added a button to the  `default.html.twig` template and a corresponding text field in the `default.yaml` blueprint.  Each page that I created now links to the content where I actually want the visitor to go.  Each page has a "robots" - "noindex" metatag so that my duplicate content is not found by google. I then modified my `simplesearch_item.html.twig` to set my href to:

```
{% if page.header.button %}{{ base_url }}{{ page.header.button }}{% else %}{{ page.url }}{% endif %}
```

Now the user finds the page I want them to find and never sees the duplicate pages. They are just for index words.

Kind of a long process, but it only took an hour and it is perfect for the end user.
